geronimo-openapi-impl/src/test/java/org/apache/geronimo/microprofile/openapi/impl/processor/SchemaProcessorTest.java

```diff
@@ -16,25 +16,24 @@
  */
 package org.apache.geronimo.microprofile.openapi.impl.processor;
 
-import static java.util.Arrays.asList;
-import static java.util.stream.Collectors.toSet;
-import static org.testng.Assert.assertEquals;
-
-import java.lang.reflect.Type;
-import java.util.List;
-import java.util.function.Supplier;
-import java.util.stream.Stream;
+import org.apache.geronimo.microprofile.openapi.impl.model.ComponentsImpl;
+import org.apache.geronimo.microprofile.openapi.openjpa.Entity1;
+import org.eclipse.microprofile.openapi.models.Components;
+import org.eclipse.microprofile.openapi.models.media.Schema;
+import org.testng.annotations.Test;
 
 import javax.json.JsonArray;
 import javax.json.JsonObject;
 import javax.json.JsonValue;
 import javax.json.bind.annotation.JsonbProperty;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
 
-import org.apache.geronimo.microprofile.openapi.impl.model.ComponentsImpl;
-import org.apache.geronimo.microprofile.openapi.openjpa.Entity1;
-import org.eclipse.microprofile.openapi.models.Components;
-import org.eclipse.microprofile.openapi.models.media.Schema;
-import org.testng.annotations.Test;
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toSet;
+import static org.testng.Assert.assertEquals;
 
 public class SchemaProcessorTest {
```

Comment:
Maybe add a test for your use case to prevent regression

Reply:
totally forgotten. I have added a line to the type test, to verify that ref is present. Also added another test to guarantee that both properties are referring the same schema.